test/cql-pytest/test_permissions.py

```diff
@@ -418,3 +418,21 @@ def test_udf_permissions_no_args(cql):
                     grant(cql, 'SELECT', table, username)
                     check_enforced(cql, username, permission='EXECUTE', resource=f'function {keyspace}.{fun}()',
                             function=lambda: user_session.execute(f'SELECT {keyspace}.{fun}() FROM {table}'))
+
+# Test that the create permission can't be granted on a single function
+def test_create_on_single_function(cql):
+    schema = "a int primary key"
+    user = "cassandra"
+    with new_test_keyspace(cql, "WITH REPLICATION = { 'class': 'SimpleStrategy', 'replication_factor': 1 }") as keyspace:
+        with new_test_table(cql, keyspace, schema) as table:
+            fun_body_lua = f"(a int, b int) CALLED ON NULL INPUT RETURNS int LANGUAGE lua AS 'return a + b;'"
+            fun_body_java = f"(a int, b int) CALLED ON NULL INPUT RETURNS int LANGUAGE java AS 'return a + b;'"
+            fun_body = fun_body_lua
+            try:
+                with new_function(cql, keyspace, fun_body):
+                    pass
+            except:
+                fun_body = fun_body_java
+            with new_function(cql, keyspace, fun_body) as fun:
+                with pytest.raises(SyntaxException):
```

Comment:
I'm surprised that this generates a SyntaxException, not a InvalidRequest. Is this what happens in Cassandra as well (i.e., did you run this test on Cassandra?).

Reply:
Yes